Yoast/wordpress-seo
# Add Alerts to PHP

## Context


Alert boxes had to be introduced to the PHP code to be able to use them in for example Health Checks. 

**Note**: Previous occurrences of `Alerts` (which were not alerts, but actually notifications) were renamed to `Notifications` in this PR: #14636 

## Summary


This PR can be summarized in the following changelog entry:

* Introduces Alert boxes to the PHP code in an Alert_Presenter class, and turns two existing Alert boxes into Alert_Presenter objects. 

## Relevant technical choices:
* Introduces a `Alert_Presenter` class. After discussing with @omarreiss it was decided to add the Alerts as a presenter in the `src` folder (in contrast to the proposed `WPSEO_Alert` class).
* Added the four icons to the `images` directory. 
* Found two occurrences of Alert boxes in the php code, in `admin/taxonomy/class-taxonomy.php` and `admin/metabox/class-metabox.php`, which were turned into `Alert_Presenter`s; the css of this code was also removed. 
* Added new css class `alerts.css`.
* Also removed some duplicate health-check css (which had to be done twice due to the changes in trunk with the removal of the sass from the plugin). 

## Test instructions

This PR can be tested by following these steps:
* Checkout this branch and build

_Test whether existing Alerts are still shown correctly_
* In `admin/metabox/class-metabox.php` :  add `return true;` to the first line of `is_internet_explorer()` 
* In the backend, go to `Posts` > `Categories` and edit a category
* The metabox should look like this:<img width="665" alt="texonomy" src="https://user-images.githubusercontent.com/43582255/78979999-429bdb80-7b1d-11ea-9804-a02a4d8df4d8.png">

* In the backend, go to `Posts` and edit a post
* The metabox should look like this: <img width="587" alt="metabox" src="https://user-images.githubusercontent.com/43582255/78980055-5f381380-7b1d-11ea-8926-4a45f2fec71f.png">

* Don't forget to remove the `return true;` from the first line of `is_internet_explorer()` 

_Check adding new Alert boxes in plugin and Health Checks_
* In `admin/views/tabs/dashboard/dashboard.php`, below `<div class="tab-block">` and above `<div class="yoast-notifications>`, add:
```
<?php
$description = sprintf(
	esc_html__( 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc sagittis vulputate neque tristique. %1$sNunc sagittis vulputate neque tristique%2$s', 'wordpress-seo' ),
	'<a href="#" target="_blank">',
	'</a>'
);
echo new Alert_Presenter( 'Simple <strong>error</strong> alert. ' . $description, 'error' );
echo new Alert_Presenter( 'Simple <strong>info</strong> alert. ' . $description, 'info' );
echo new Alert_Presenter( 'Simple <strong>success</strong> alert. ' . $description, 'success' );
echo new Alert_Presenter( 'Simple <strong>warning</strong> alert. ' . $description );
?>
```
* Also add `use Yoast\WP\SEO\Presenters\Admin\Alert_Presenter;` on the top of the file
* In the backend, go to `SEO` > `General`, which should show:
<img width="1916" alt="Screenshot 2020-03-31 at 13 58 58" src="https://user-images.githubusercontent.com/43582255/78980505-50059580-7b1e-11ea-9b79-599820f3f1dd.png">



* In one of the health checks `inc/health-check-default-tagline.php`
* Go to `Settings` > `Reading`: make sure the `Tagline` is “Just another WordPress site” (e.g. the default tagline)
* Below `$this->description = esc_html__( 'You still have the default WordPress tagline. Even an empty one is probably better.', 'wordpress-seo' );`, paste :

```
$description = sprintf(
	esc_html__( 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc sagittis vulputate neque tristique. %1$sNunc sagittis vulputate neque tristique%2$s', 'wordpress-seo' ),
	'<a href="#" target="_blank">',
	'</a>'
);
$alert = new Alert_Presenter( 'Simple <strong>error</strong> alert. ' . $description, 'error' );
$this->description .= $alert->present();
$alert = new Alert_Presenter( 'Simple <strong>info</strong> alert. ' . $description, 'info' );
$this->description .= $alert->present();
$alert = new Alert_Presenter( 'Simple <strong>success</strong> alert. ' . $description, 'success' );
$this->description .= $alert->present();
$alert = new Alert_Presenter( 'Simple <strong>warning</strong> alert. ' . $description );
$this->description .= $alert->present();
```
* Also add `use Yoast\WP\SEO\Presenters\Admin\Alert_Presenter;` on the top of the file
* In the backend, go to `Tools` > `Site Health`, which should show:
<img width="641" alt="Screenshot 2020-03-31 at 13 58 31" src="https://user-images.githubusercontent.com/43582255/78980437-32383080-7b1e-11ea-9d1d-a9580d53f36a.png">


## UI changes

## Documentation

## Quality assurance


Fixes #14585
